packages/rest/src/events/WebSocketEvents.ts

```diff
@@ -0,0 +1,95 @@
+import type { ServerConfig } from '../utils/ServerConfig'
+import type {
+  Agent,
+  BasicMessageStateChangedEvent,
+  ConnectionStateChangedEvent,
+  CredentialStateChangedEvent,
+  ProofStateChangedEvent,
+} from '@aries-framework/core'
+
+import {
+  BasicMessageEventTypes,
+  ConnectionEventTypes,
+  CredentialEventTypes,
+  ProofEventTypes,
+} from '@aries-framework/core'
+import WebSocket from 'ws'
+
+import { sendWebhookEvent } from './WebhookEvent'
+
+export const webSocketEvents = async (
+  server: WebSocket.Server,
+  stream: WebSocket,
+  agent: Agent,
+  config: ServerConfig
+) => {
+  stream.on('message', function message(data) {
+    server.clients.forEach((client) => {
+      if (client.readyState === WebSocket.OPEN) {
+        client.send(data)
+      }
+    })
+  })
+
+  agent.events.on(
```

Comment:
We already have a listener for the webhook events, I think we can reuse those listener also for the websocket events (in the other events files).

Also, would this mean a webhook is sent twice to the webhookUrl now?